### DISCUSSION TITLE
What is the binary vector required for Hamming distance? Any examples?

### DISCUSSION THREAD

**Discussion by ingale726** [milvus-io/milvus#23585] [Q&A and General discussion]
`[0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 1]`

When I insert data like this, it tells me I should insert a binary vector instead of a float vector

**Comment by ingale726** [comment]
```
from bitarray import bitarray

binary_vector = bitarray('0101010101010101010101010101010101010101010101010101010101010101')
```
Can a vector like this be inserted into a database?

**Comment by yhmo** [comment]
https://github.com/milvus-io/pymilvus/blob/master/examples/collection.py

```
 raw_vector = [random.randint(0, 1) for i in range(dim)]
 binary_vectors.append(bytes(np.packbits(raw_vector, axis=-1).tolist()))
```
